bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/onvif/OnvifDiscovery.java

```diff
@@ -213,26 +215,21 @@ protected void channelRead0(@Nullable ChannelHandlerContext ctx, DatagramPacket
                 }).option(ChannelOption.SO_BROADCAST, true).option(ChannelOption.SO_REUSEADDR, true)
                 .option(ChannelOption.IP_MULTICAST_LOOP_DISABLED, false).option(ChannelOption.SO_RCVBUF, 2048)
                 .option(ChannelOption.IP_MULTICAST_TTL, 255).option(ChannelOption.IP_MULTICAST_IF, networkInterface);
-
-        datagramChannel = (DatagramChannel) bootstrap.bind(localNetworkAddress).sync().channel();
-        datagramChannel.joinGroup(multiCastAddress, networkInterface).sync();
-        ChannelFuture chFuture;
-        if (port == 1900) {
-            String ssdp = "M-SEARCH * HTTP/1.1\n" + "HOST: 239.255.255.250:1900\n" + "MAN: \"ssdp:discover\"\n"
-                    + "MX: 1\n" + "ST: urn:dial-multiscreen-org:service:dial:1\n"
-                    + "USER-AGENT: Microsoft Edge/83.0.478.61 Windows\n" + "\n" + "";
-            ByteBuf ssdpProbeMessage = Unpooled.copiedBuffer(ssdp, 0, ssdp.length(), StandardCharsets.UTF_8);
-            datagramPacket = new DatagramPacket(ssdpProbeMessage, multiCastAddress, localNetworkAddress);
-            chFuture = datagramChannel.writeAndFlush(datagramPacket);
-        } else {
-            chFuture = datagramChannel.writeAndFlush(datagramPacket);
+        ChannelGroup openChannels = new DefaultChannelGroup(GlobalEventExecutor.INSTANCE);
+        for (NetworkInterface nic : nics) {
+            DatagramChannel datagramChannel = (DatagramChannel) bootstrap.option(ChannelOption.IP_MULTICAST_IF, nic)
+                    .bind(new InetSocketAddress(0)).sync().channel();
+            datagramChannel
+                    .joinGroup(new InetSocketAddress(InetAddress.getByName("239.255.255.250"), 3702), networkInterface)
+                    .sync();
+            openChannels.add(datagramChannel);
+        }
+        if (!openChannels.isEmpty()) {
+            openChannels.writeAndFlush(wsDiscovery());
```

Comment:
Does this still return a future that you should wait for? Or do you not need to do that anymore?

Reply:
I don’t need to use a future as it gives no benefits and only increases code length.